src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/UnnamedSymbolCompletionProvider.cs

```diff
@@ -0,0 +1,208 @@
+
+        /// <summary>
+        /// We keep operators sorted in a specific order.  We don't want to sort them alphabetically, but instead want
+        /// to keep things like <c>==</c> and <c>!=</c> together.
+        /// </summary>
+        private static string SortText(int sortingGroupIndex, string sortTextSymbolPart)
+            => $"{SortingPrefix}{sortingGroupIndex:000}_{sortTextSymbolPart}";
+
+        /// <summary>
+        /// Gets the dot-like token we're after, and also the start of the expression we'd want to place any text before.
+        /// </summary>
+        private static (SyntaxToken dotLikeToken, int expressionStart) GetDotAndExpressionStart(SyntaxNode root, int position)
+        {
+            var tokenOnLeft = root.FindTokenOnLeftOfPosition(position, includeSkipped: true);
+            var dotToken = tokenOnLeft.GetPreviousTokenIfTouchingWord(position);
+
+            // Has to be a . or a .. token
+            if (!CompletionUtilities.TreatAsDot(dotToken, position - 1))
+                return default;
+
+            // don't want to trigger after a number.  All other cases after dot are ok.
+            if (dotToken.GetPreviousToken().Kind() == SyntaxKind.NumericLiteralToken)
+                return default;
+
+            // if we have `.Name`, we want to get the parent member-access of that to find the starting position.
+            // Otherwise, if we have .. then we want the left side of that to find the starting position.
+            var expression = dotToken.Kind() == SyntaxKind.DotToken
+                ? dotToken.Parent as ExpressionSyntax
+                : (dotToken.Parent as RangeExpressionSyntax)?.LeftOperand;
+
+            if (expression == null)
+                return default;
+
+            // If we're after a ?. find the root of that conditional to find the start position of the expression.
+            expression = expression.GetRootConditionalAccessExpression() ?? expression;
+            return (dotToken, expression.SpanStart);
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var dotAndExprStart = GetDotAndExpressionStart(root, position);
+            if (dotAndExprStart == default)
+                return;
+
+            var recommender = document.GetRequiredLanguageService<IRecommendationService>();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var options = CodeAnalysis.Completion.Providers.CompletionUtilities.GetUpdatedRecommendationOptions(context.Options, document.Project.Language);
+            var recommendedSymbols = recommender.GetRecommendedSymbolsAtPosition(
```

Comment:
@CyrusNajmabadi We are calling recommender twice for each completion session (here and symbol provider), will there still be any extra cost for the 2nd call or there's cache in compiler that takes care of this?

Reply:
in my measurements this was cheap as all the symbols were there from teh first call.  all we're doing is just walking teh type (and base types) again and just getting these members.  as they're all in memory, it's practically free.  note that we're not doing htings like import completion again.  just symbols.